### PR TITLE
Settings service

### PR DESCRIPTION
**Pull request by romaroid**

### CODE REVIEW

**Comment by rubensayshi** [comment]
checklist for myself for review process
- [ ] review settings service code
- [ ] review settings ctrl code
- [ ] test AppVersionService
- [ ] test settings
- [ ] test glidera

File: src/js/controllers/WalletControllers.js [hunk 1]
```diff
@@ -54,17 +61,20 @@ angular.module('blocktrail.wallet')
...
-                            settingsService.$store().then(function() {
-                                settingsService.$syncSettingsUp();
-                            });
+                            var updateSettings = {
+                                latestVersionWeb: CONFIG.VERSION
+                            };
+
+                            settingsService.updateSettingsUp(updateSettings);
+                                // Check this logic
+                                // .then(checkGlideraActivated);
```
Comment (rubensayshi) [inline src/js/controllers/WalletControllers.js]: old commented code, should this be enabled or removed?

File: src/js/modules/wallet/controllers/settings/settings.tpl.html [hunk 2]
```diff
@@ -0,0 +1,218 @@
...
+                                        style="background-image: url('{{ formSettings.profilePic }}');"
+                                ></span>
+                                </div>
+
+                                <div class="btn-file-data-holder iit-10">
+                                    <btn-file-data
+                                            btn-file-data-name="'profileIcon'"
+                                            btn-file-data-update="onFileDataUpdate(name, data)"
+                                    >
+                                        Upload file
```
Comment (rubensayshi) [inline src/js/modules/wallet/controllers/settings/settings.tpl.html]: needs translation

File: src/js/services/GlideraService.js [hunk 3]
```diff
@@ -402,14 +412,11 @@ angular.module('blocktrail.wallet').factory(
...
-                                    });
-
-                                    return settingsService.$store().then(function() {
-                                        return settingsService.$syncSettingsUp().then(function() {
-                                            updatePendingTransactions();
+                                    };
 
-                                            return result;
-                                        });
+                                    return settingsService.addGlideraTransaction().then(function() {
```
Comment (rubensayshi) [inline src/js/services/GlideraService.js]: missing argument

File: src/js/modules/core/services/settings/settings.service.js [hunk 4]
```diff
@@ -0,0 +1,585 @@
...
+     * @param propertyList
+     * @returns {boolean}
+     */
+    Settings.prototype.checkOnPropertyUpList = function(obj, propertyList) {
+        var flag = false;
+        var i = 0;
+
+        while(i < propertyList.length && !flag) {
+            if(angular.isDefined(obj[propertyList[i]])) {
+                flag = true;
```
Comment (rubensayshi) [inline src/js/modules/core/services/settings/settings.service.js]: should just `return true` here

File: src/js/modules/core/services/settings/settings.service.js [hunk 5]
```diff
@@ -0,0 +1,585 @@
...
+        }
+
+        return promise;
+    };
+
+    /**
+     * Get settings object return a link on _readonlyDoc
+     * @returns _readonlyDoc {promise}
+     */
+    Settings.prototype.getSettingsObject = function() {
```
Comment (rubensayshi) [inline src/js/modules/core/services/settings/settings.service.js]: should we just call this `getReadOnlySettings` ?

**Comment by rubensayshi** [comment]
`@import "helpers/variables";` needs to be before `@import "../lib/bootstrap-sass/assets/stylesheets/_bootstrap";` because some of those variables adjust the way bootstrap is.

for example the body background is set by setting `$body-bg`, same for some of the font options etc.

**Comment by rubensayshi** [comment]
all the urls in the `fonts/*.scss` should be `../font`, not `../../font`, the path is relative to the bundled css file

**Comment by rubensayshi** [comment]
you missed 2 `settings.glideraAccessToken` replacements in `GlideraService.js`

File: src/js/modules/core/services/settings/settings.service.js (hunk 4 above)
Comment (romaroid) [inline src/js/modules/core/services/settings/settings.service.js]: I prefer to have one return in the function

**Pull request merged**